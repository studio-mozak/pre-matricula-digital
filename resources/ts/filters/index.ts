import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
