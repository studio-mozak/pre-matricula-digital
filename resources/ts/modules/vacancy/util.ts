import { VacancyLoadProcess, VacancyLoadProcessAnalyse } from './types';

export const trunc = (value: number) => Math.trunc(value * 10000) / 100;

export const analyse = (p: VacancyLoadProcess): VacancyLoadProcessAnalyse => {
  const process = {
    ...p,
    unavailable: 0,
    exceded: 0,
    ultrapassed: 0,
    percentual: {
      accepted: 0,
      available: 0,
      unavailable: 0,
      exceded: 0,
      totalExceded: 0,
      total: 0,
      waiting: 0,
      waitingRealAvailable: 0,
      waitingExceded: 0,
      waitingAvailable: 0,
    },
  } as VacancyLoadProcessAnalyse;

  // Se o número de vagas disponíveis for negativo, quer dizer que as vagas foram excedidas
  if (process.available < 0) {
    process.exceded = Math.abs(process.available);
    process.available = 0;
  }

  // Se o número de vagas aceitas for maior que o total de vagas, quer dizer que os limites foram ultrapassados
  if (process.accepted > process.total) {
    process.ultrapassed = process.accepted - process.total;
  }

  process.unavailable = process.total - process.accepted - process.available;
  process.max = process.total + process.exceded + process.ultrapassed;
  process.totalExceded = process.exceded + process.ultrapassed;

  process.percentual.total = trunc(process.total / process.max);
  process.percentual.totalExceded = trunc(process.totalExceded / process.max);
  process.percentual.accepted = trunc(process.accepted / process.max);
  process.percentual.available = trunc(process.available / process.max);
  process.percentual.unavailable = trunc(process.unavailable / process.max);
  process.percentual.exceded = trunc(process.exceded / process.max);

  process.waitingAvailable = process.available - process.waiting;
  process.waitingExceded = 0;

  if (process.waitingAvailable < 0) {
    process.waitingExceded = Math.abs(process.waitingAvailable);
    process.waitingAvailable = 0;
  }

  process.maxWaiting = Math.max(process.available, process.waiting);
  process.percentual.waiting = trunc(process.waiting / process.maxWaiting);
  process.percentual.waitingRealAvailable = trunc(
    process.waitingAvailable / process.maxWaiting
  );
  process.percentual.waitingAvailable = trunc(
    process.available / process.maxWaiting
  );
  process.percentual.waitingExceded = trunc(
    process.waitingExceded / process.maxWaiting
  );

  return process;
};
