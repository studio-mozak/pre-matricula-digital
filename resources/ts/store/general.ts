import { Course, Grade, Period, SchoolYear } from '@/modules/processes/types';
import { Nullable, Option, Processes } from '@/types';
import { School } from '@/modules/school/types';
import axios from 'axios';
import { defineStore } from 'pinia';

declare global {
  interface Window {
    config: {
      city: string;
      state: string;
      ibge_codes: string;
      map: {
        lat: number;
        lng: number;
        zoom: number;
      };
      token: string;
      logo: string;
      slogan: string;
      allow_optional_address: boolean;
      show_how_to_do_video: boolean;
      video_intro_url: string;
      link_to_restrict_area: string;
      features: {
        allow_preregistration_data_update: boolean;
        allow_external_system_data_update: boolean;
        allow_transfer_registration: boolean;
        transfer_description: string;
        allow_vacancy_certificate: boolean;
      };
    };
  }
}

type MapCenter = {
  lat: number;
  lng: number;
};

type MapConfigStyleStyler = {
  visibility: string;
};

type MapConfigStyle = {
  featureType: string;
  stylers: MapConfigStyleStyler[];
};

type MapConfig = {
  clickableIcons: boolean;
  streetViewControl: boolean;
  panControlOptions: boolean;
  mapTypeControl: boolean;
  styles: MapConfigStyle[];
};

type Map = {
  config: MapConfig;
  center: MapCenter;
  zoom: number;
};

type Entity = {
  city: string;
  state: string;
  ibgeCodes: number[];
};

type Config = {
  ['allow_optional_address']: boolean;
  ['show_how_to_do_video']: boolean;
  ['video_intro_url']: string;
  ['link_to_restrict_area']: string;
};

export type User = {
  name: string;
  level: number;
  schools: string[];
};

type Auth = {
  authenticated: boolean;
  checked: boolean;
  user: Nullable<User>;
};

interface GeneralState {
  auth: Auth;
  map: Map;
  entity: Entity;
  logo: string;
  slogan: string;
  config: Config;
  features: {
    allowPreregistrationDataUpdate: boolean;
    allowExternalSystemDataUpdate: boolean;
    allowTransferRegistration: boolean;
    transferDescription: string;
    allowVacancyCertificate: boolean;
  };
  cities: Option[];
  stageTypes: Option[];
  relationTypes: Option[];
  restrictionTypes: Option[];
  processes: Processes[];
  schools: School[];
  courses: Course[];
  grades: Grade[];
  periods: Period[];
  schoolYears: SchoolYear[];
}

export const useGeneralStore = defineStore('general', {
  state: (): GeneralState => ({
    auth: {
      authenticated: false,
      checked: false,
      user: null,
    },
    map: {
      config: {
        clickableIcons: false,
        streetViewControl: false,
        panControlOptions: false,
        mapTypeControl: false,
        styles: [
          {
            featureType: 'poi',
            stylers: [{ visibility: 'off' }],
          },
        ],
      },
      center: {
        lat: window.config.map.lat,
        lng: window.config.map.lng,
      },
      zoom: window.config.map.zoom,
    },
    entity: {
      city: window.config.city,
      state: window.config.state,
      ibgeCodes: window.config.ibge_codes
        .split(',')
        .filter((f: string) => f)
        .map((m: string) => Number(m)),
    },
    logo: window.config.logo,
    slogan: window.config.slogan,
    config: {
      allow_optional_address: window.config.allow_optional_address,
      show_how_to_do_video: window.config.show_how_to_do_video,
      video_intro_url: window.config.video_intro_url,
      link_to_restrict_area: window.config.link_to_restrict_area,
    },
    features: {
      allowPreregistrationDataUpdate:
        window.config.features.allow_preregistration_data_update,
      allowExternalSystemDataUpdate:
        window.config.features.allow_external_system_data_update,
      allowTransferRegistration:
        window.config.features.allow_transfer_registration,
      transferDescription: window.config.features.transfer_description,
      allowVacancyCertificate: window.config.features.allow_vacancy_certificate,
    },
    processes: [],
    schools: [],
    courses: [],
    grades: [],
    periods: [],
    schoolYears: [],
    stageTypes: [
      {
        key: 'REGISTRATION',
        label: 'Matrícula',
      },
      {
        key: 'REGISTRATION_RENEWAL',
        label: 'Rematrícula',
      },
      {
        key: 'WAITING_LIST',
        label: 'Lista de espera',
      },
    ],
    relationTypes: [
      {
        key: 'MOTHER',
        label: 'Mãe',
      },
      {
        key: 'FATHER',
        label: 'Pai',
      },
      {
        key: 'GUARDIAN',
        label: 'Responsável Legal',
      },
      {
        key: 'SELF',
        label: 'O próprio (você é o(a) aluno(a))',
      },
    ],
    restrictionTypes: [
      {
        key: 'NONE',
        label: 'Todas',
        subLabel: 'Não haverá restrição para se inscrever',
      },
      {
        key: 'REGISTRATION_LAST_YEAR',
        label: 'Matrícula ativa no ano anterior ao processo',
        subLabel:
          'Apenas alunos(as) com matrícula ativa ao ano anterior do processo poderão se inscrever (rematrícula)',
      },
      {
        key: 'REGISTRATION_CURRENT_YEAR',
        label: 'Matrícula ativa no ano do processo',
        subLabel:
          'Apenas alunos(as) com matrícula ativa no ano do processo poderão se inscrever (remanejamentos/trocas)',
      },
      {
        key: 'NO_REGISTRATION_CURRENT_YEAR',
        label: 'Sem matrícula ativa no ano do processo',
        subLabel:
          'Apenas alunos(as) sem matrícula ativa no ano do processo poderão se inscrever (retorno)',
      },
      {
        key: 'NO_REGISTRATION_PERIOD_CURRENT_YEAR',
        label: 'Sem matrícula ativa no turno e ano do processo',
        subLabel:
          'Apenas alunos(as) sem matrícula ativa no(s) turno(s) e ano do processo poderão se inscrever',
      },
      {
        key: 'NEW_STUDENT',
        label: 'Novos alunos',
        subLabel:
          'Apenas alunos(as) sem cadastro no sistema poderão se inscrever (novo aluno)',
      },
    ],
    cities: [],
  }),
  getters: {
    isAdmin: (state: GeneralState) => state.auth.user?.level === 1,
    isCoordinator: (state: GeneralState) => state.auth.user?.level === 2,
    isSecretary: (state: GeneralState) => state.auth.user?.level === 4,
    getUserSchools: (state: GeneralState) =>
      state.auth.user ? state.auth.user.schools : [],
    getStageTypes: (state: GeneralState) => state.stageTypes,
    getRestrictionTypes: (state: GeneralState) => state.restrictionTypes,
    isProduction: () => process.env.NODE_ENV === 'production',
    isAuthenticated: (state: GeneralState) => state.auth.authenticated,
    getAuthenticatedUser: (state: GeneralState) => state.auth.user,
    getEntityData: (state: GeneralState) => state.entity,
    getSlogan: (state: GeneralState) => state.slogan,
    getLogo: (state: GeneralState) => state.logo,
    getCities: (state: GeneralState) => state.cities,
    getConfig: (state: GeneralState) => state.config,
  },
  actions: {
    authenticated(user: User) {
      this.auth.authenticated = true;
      this.auth.checked = true;
      this.auth.user = user;
    },
    checked() {
      this.auth.checked = true;
    },
    setProcesses(payload: Processes[]) {
      this.processes = payload;
    },
    setSchools(payload: School[]) {
      this.schools = payload;
    },
    addCities(payload: Option[]) {
      const cities = this.cities.concat(payload);
      // Adiciona a cidade no array de cidades, removendo duplicatas
      this.cities = cities.filter(
        (obj, key, array) =>
          key ===
          array.findIndex(
            (arr) => arr.key === obj.key && arr.label === obj.label
          )
      );
    },
    check() {
      return axios.get('/auth/check').then(({ data }) => {
        this.authenticated(data);
      });
    },
  },
  persist: false,
});
