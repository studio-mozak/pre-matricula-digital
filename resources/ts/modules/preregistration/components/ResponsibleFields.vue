<template>
  <div>
    <h2 class="font-muli-20-primary">
      Preencha o formulário abaixo para criar sua inscrição:
    </h2>
    <p>
      Preencha os campos abaixo para identificar os dados do(a) responsável pela
      inscrição do(a) aluno(a). Os campos com asterisco (*) são de preenchimento
      <strong>obrigatório</strong> para prosseguir com a inscrição.
    </p>
    <hr class="mt-5 mb-5" />
    <div class="mb-3 d-sm-flex justify-content-between">
      <h3 class="font-hind-18-primary">
        Dados do(a) responsável pelo(a) aluno(a)
      </h3>
      <x-btn
        label="Limpar dados"
        class="border-danger text-danger flex-row"
        no-caps
        no-wrap
        @click="$emit('clean')"
      />
    </div>
    <fields
      v-model:fields="modelFields"
      v-model:data="modelResponsible"
      :errors="errors"
    />
    <p class="mt-4 mb-4">Vínculo com o(a) aluno(a)</p>
    <div class="row">
      <x-field
        v-model="modelResponsible['relationType']"
        container-class="col-12 col-md-6 form-group"
        rules="required"
        label="Relação com o(a) aluno(a)"
        name="relationType"
        type="SELECT"
        :options="store.relationTypes"
        :errors="!!errors['relationType']"
      />
    </div>
    <p class="mt-4 mb-4">Endereço residencial</p>
    <p class="text-muted">
      Digite o <strong>CEP</strong> que corresponde ao seu endereço e clique em
      'Buscar' para preenchimento automático. Caso seu endereço não seja
      encontrado, preencha manualmente os campos necessários para identificá-lo
      (endereço, número, bairro).
    </p>
    <address-fields
      v-model:fetchingAddressLatLng="modelFetchingPrimaryAddressLatLng"
      v-model:data="modelResponsible.address"
      :set-field-value="setFieldValue"
      :errors="errors"
      name="address"
    />
    <div
      v-if="
        modelResponsible &&
        store.config.allow_optional_address &&
        !modelResponsible.useSecondAddress
      "
      class="row mt-5"
    >
      <div class="col-12 col-md-6 offset-md-3">
        <x-btn
          data-test="btn-use-second-address"
          label="Adicionar endereço secundário"
          color="primary"
          class="w-100"
          no-caps
          no-wrap
          @click="modelResponsible.useSecondAddress = true"
        />
      </div>
    </div>
    <template v-if="modelResponsible && modelResponsible.useSecondAddress">
      <p class="mt-4 mb-4">Endereço secundário</p>
      <address-fields
        v-model:data="modelResponsible.secondAddress"
        v-model:fetchingAddressLatLng="modelFetchingSecondaryAddressLatLng"
        :set-field-value="setFieldValue"
        :errors="errors"
        name="secondAddress"
      />
      <div class="row mt-5">
        <div class="col-12 col-md-6 offset-md-3">
          <x-btn
            data-test="btn-doenst-use-second-address"
            label="Não usar endereço secundário"
            class="border-danger text-danger w-100"
            no-caps
            no-wrap
            @click="doesntuseSecondAddress"
          />
        </div>
      </div>
    </template>
  </div>
</template>

<script lang="ts">
import {
  ParseFieldFromProcess,
  PreRegistrationResponsibleField,
} from '../types';
import { PropType, defineComponent } from 'vue';
import AddressFields from '@/components/form/AddressFields.vue';
import { GenericFields } from '@/components/form/Fields.vue';
import XBtn from '@/components/elements/buttons/XBtn.vue';
import XField from '@/components/x-form/XField.vue';
import { newAddress } from '@/util';
import { useGeneralStore } from '@/store/general';
import { useVModel } from '@vueuse/core';

export default defineComponent({
  components: {
    AddressFields,
    Fields: GenericFields<PreRegistrationResponsibleField>(),
    XBtn,
    XField,
  },
  props: {
    responsible: {
      type: Object as PropType<PreRegistrationResponsibleField>,
      default: () => ({}),
    },
    errors: {
      type: Object as PropType<{
        [key: string]: boolean;
      }>,
      default: () => ({}),
    },
    fields: {
      type: Array as PropType<ParseFieldFromProcess[]>,
      default: () => [],
    },
    fetchingPrimaryAddressLatLng: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    fetchingSecondaryAddressLatLng: {
      type: Boolean as PropType<boolean>,
      default: false,
    },
    setFieldValue: {
      type: Function as PropType<
        (field: string, value: string | number) => void
      >,
      required: false,
      default: () => ({}),
    },
  },
  emits: [
    'clean',
    'update:fetchingPrimaryAddressLatLng',
    'update:fetchingSecondaryAddressLatLng',
  ],
  setup(props) {
    const store = useGeneralStore();

    const modelFields = useVModel(props, 'fields');
    const modelResponsible = useVModel(props, 'responsible');
    const modelFetchingPrimaryAddressLatLng = useVModel(
      props,
      'fetchingPrimaryAddressLatLng'
    );
    const modelFetchingSecondaryAddressLatLng = useVModel(
      props,
      'fetchingSecondaryAddressLatLng'
    );

    const doesntuseSecondAddress = () => {
      modelResponsible.value.useSecondAddress = false;
      modelResponsible.value.secondAddress = newAddress();
    };

    return {
      store,
      modelFields,
      modelResponsible,
      modelFetchingPrimaryAddressLatLng,
      modelFetchingSecondaryAddressLatLng,
      doesntuseSecondAddress,
    };
  },
});
</script>
