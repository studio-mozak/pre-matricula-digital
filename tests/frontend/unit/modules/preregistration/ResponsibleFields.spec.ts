import {
  ParseFieldFromProcess,
  PreRegistrationResponsibleField,
} from '@/modules/preregistration/types';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import ResponsibleFields from '@/modules/preregistration/components/ResponsibleFields.vue';
import VTooltip from 'v-tooltip';
import { createTestingPinia } from '@pinia/testing';
import filters from '@/filters';
import { newAddress } from '@/util';
import { useGeneralStore } from '@/store/general';
import { expect, vi } from 'vitest';

interface ResponsibleFieldsProps {
  responsible: PreRegistrationResponsibleField;
  errors: {
    [key: string]: boolean;
  };
  fields: ParseFieldFromProcess[];
  fetchingPrimaryAddressLatLng: boolean;
  fetchingSecondaryAddressLatLng: boolean;
  setFieldValue: (field: string, value: string | number) => void;
}

const propsMock: ResponsibleFieldsProps = {
  responsible: {
    relationType: null,
    address: {
      postalCode: '',
      address: '',
      number: '',
      complement: '',
      neighborhood: '',
      lat: 0,
      lng: 0,
      city: '',
      cityIbgeCode: 0,
      stateAbbreviation: '',
    },
    useSecondAddress: false,
    secondAddress: {
      postalCode: '',
      address: '',
      number: '',
      complement: '',
      neighborhood: '',
      lat: 0,
      lng: 0,
      city: '',
      cityIbgeCode: 0,
      stateAbbreviation: '',
    },
    responsible_name: null,
  } as PreRegistrationResponsibleField,
  errors: {},
  fields: [],
  fetchingPrimaryAddressLatLng: false,
  fetchingSecondaryAddressLatLng: false,
  setFieldValue: () => {
    return;
  },
};

const onMounted = async (props: ResponsibleFieldsProps) => {
  const wrapper = mount(ResponsibleFields, {
    props,
    global: {
      plugins: [createTestingPinia()],
      stubs: ['x-field', 'address-fields'],
      directives: {
        tooltip: VTooltip,
      },
      provide: {
        $filters: filters,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  const store = useGeneralStore();

  store.config.allow_optional_address = true;

  return wrapper;
};

describe('ResponsibleFields', () => {
  let wrapper: VueWrapper<InstanceType<typeof ResponsibleFields>>;

  beforeEach(async () => {
    wrapper = await onMounted(propsMock);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('useSecondAddress', () => {
    beforeEach(async () => {
      wrapper = await onMounted(propsMock);

      await wrapper.vm.$nextTick();
    });

    afterEach(() => {
      vi.clearAllMocks();
    });

    test('when I click in use second address button, modelResponsible.useSecondAddress must be set to true', async () => {
      wrapper.vm.modelResponsible.useSecondAddress = false;

      expect(wrapper.vm.modelResponsible.useSecondAddress).toBeFalsy();

      const btn = wrapper.find('[data-test="btn-use-second-address"]');

      await btn.trigger('click');

      expect(wrapper.vm.modelResponsible.useSecondAddress).toBeTruthy();
    });

    test("when I click in doesn't use second address button, the doesntuseSecondAddress function must be called", async () => {
      wrapper.vm.modelResponsible.useSecondAddress = true;

      expect(wrapper.vm.modelResponsible.useSecondAddress).toBeTruthy();

      await wrapper.vm.$nextTick();

      const btn = wrapper.find('[data-test="btn-doenst-use-second-address"]');

      await btn.trigger('click');

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.responsible.useSecondAddress).toBe(false);
      expect(wrapper.vm.responsible.secondAddress).toMatchObject(newAddress());
    });
  });
});
