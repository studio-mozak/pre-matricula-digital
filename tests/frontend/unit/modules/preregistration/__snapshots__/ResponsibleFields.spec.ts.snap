// Vitest Snapshot v1

exports[`ResponsibleFields > should match snapshot 1`] = `
<div>
  <h2
    class="font-muli-20-primary"
  >
     Preencha o formulário abaixo para criar sua inscrição: 
  </h2>
  <p>
     Preencha os campos abaixo para identificar os dados do(a) responsável pela inscrição do(a) aluno(a). Os campos com asterisco (*) são de preenchimento 
    <strong>
      obrigatório
    </strong>
     para prosseguir com a inscrição. 
  </p>
  <hr
    class="mt-5 mb-5"
  />
  <div
    class="mb-3 d-sm-flex justify-content-between"
  >
    <h3
      class="font-hind-18-primary"
    >
       Dados do(a) responsável pelo(a) aluno(a) 
    </h3>
    <button
      class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase border-danger text-danger flex-row"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
      >
        <span
          class="block"
        >
          Limpar dados
        </span>
      </span>
      <transition-stub
        appear="false"
        css="true"
        name="x-transition--fade"
        persisted="false"
      >
        <!---->
      </transition-stub>
    </button>
  </div>
  <div
    class="row"
  >
    
    
  </div>
  <p
    class="mt-4 mb-4"
  >
    Vínculo com o(a) aluno(a)
  </p>
  <div
    class="row"
  >
    <x-field-stub
      allowclear="false"
      bind="[object Object]"
      containerclass="col-12 col-md-6 form-group"
      disabled="false"
      errors="false"
      label="Relação com o(a) aluno(a)"
      loading="false"
      name="relationType"
      options="[object Object],[object Object],[object Object],[object Object]"
      rules="required"
      searchable="false"
      type="SELECT"
    />
  </div>
  <p
    class="mt-4 mb-4"
  >
    Endereço residencial
  </p>
  <p
    class="text-muted"
  >
     Digite o 
    <strong>
      CEP
    </strong>
     que corresponde ao seu endereço e clique em 'Buscar' para preenchimento automático. Caso seu endereço não seja encontrado, preencha manualmente os campos necessários para identificá-lo (endereço, número, bairro). 
  </p>
  <address-fields-stub
    data="[object Object]"
    errors="[object Object]"
    fetchingaddresslatlng="false"
    name="address"
    setfieldvalue="[Function]"
  />
  <div
    class="row mt-5"
  >
    <div
      class="col-12 col-md-6 offset-md-3"
    >
      <button
        class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
        data-test="btn-use-second-address"
        role="button"
        tabindex="0"
        type="button"
      >
        <span
          class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
        >
          <span
            class="block"
          >
            Adicionar endereço secundário
          </span>
        </span>
        <transition-stub
          appear="false"
          css="true"
          name="x-transition--fade"
          persisted="false"
        >
          <!---->
        </transition-stub>
      </button>
    </div>
  </div>
  <!--v-if-->
</div>
`;
